tests/fail/extern-so/function_not_in_SO.rs

```diff
@@ -1,6 +1,6 @@
 //@only-target-linux
 //@only-on-host
-//@compile-flags: -Zmiri-extern-so-file=tests/extern-so/libtestlib.so
+//@compile-flags: -Zmiri-extern-so-file=target/miri-extern-so/libtestlib.so
```

Comment:
Uh, I don't think this will work in x.py when the target dir is set to somewhere else.

Reply:
Ah good point -- is there a handy regex I can use to get the target dir here, or should I add the flag through `compiletests.rs`?

Reply:
Hm, I am honestly not sure what the best way is to do this. If you add the flag in `compiletest.rs` it will be added to *all* tests.

Reply:
That's true, hmm. One option is I could try and make a list of the tests that need it and add it only to those in `compiletest.rs`. The nicest option would be if there was a macro I could use in the compile flags, like `$TARGET_DIR` or something, but that might be a pain to implement.

Reply:
The best option I can think of for now is to have a separate folder of tests, `tests/pass-extern-so`, and have compiletest pass the right flag to all those tests.

Reply:
ah, sounds good!

Reply:
Now we have two test suites with just one test each... still the best I can think of for now. Let's see if @oli-obk can come up with anything better.

Reply:
yea that's the best for now. We can migrate to something better later